### ISSUE TITLE
Typescript: Some getter methods return void

### ISSUE THREAD

**Issue by Abbo44** [highcharts/highcharts#10083]
#### Expected behaviour
I want the Serie color by calling `getColor()`, the Serie symbol by calling `getSymbol()`

#### Actual behaviour
Those methods return a void type.
Some naming methods are weird, because when I call `getExtremes(yData?: Array<number>): void;` and `getChartSize()`, I expect to have a return but in the method docs it says for example for `getChartSize()`

> Sets Chart.chartWidth and Chart.chartHeight.

#### Product version
Highcharts 7.0.2

**Comment by KacperMadej** [comment]
Hi @Abbo44 

`getColor` gets the color and sets it on the series (the series is the function's context), so TS docs are right, but general documentation is not clear enough and the name is surely misleading. The same applies to `getSymbol` and Series' `getExtremes`. Each of those functions is more like a setter than getter.

Those functions names are wrong or those functions should get and return something as the name suggests.

@TorsteinHonsi Could this be resolved without a breaking change?

**Comment by TorsteinHonsi** [comment]
Yes I believe so. I agree it is bad naming. Then again `setColor` may not be much better, because with a name like that, I would expect that I could supply an argument and set the color to that value.

What these functions have in common is that they compute/resolve a value based on some state and sets it as properties.

What about `resolveColor`, `resolveSymbol` and `resolveExtremes`?

Or are there other more suiting verbs? Apply?

**Comment by bre1470** [comment]
My suggestions:
init... `initColor`, `initSymbol` and `initExtremes`
setup... `setupColor`, `setupSymbol` and `setupExtremes`

**Comment by pawelfus** [comment]
We already have some of that:
- `applySomething`, for example `applyZones` (Zones)
- `initSomething`, for example `initEvents` (Scrollbar), `initAnnotation` (Annotations), `initNavigationBindings` (Bindings), or `initSeries` (Chart)

We can also try:
- `pickSomething` - `pickSymbol`, `pickColor`, `pickExtremes` (?)
- `selectSomething`- `selectSymbol`, `selectColor`, `selectExtremes` (?)
- `computeSomething`- `computeSymbol` (?), `computeColor` (?), `computeExtremes`

Personally `initSomething` sounds really good. However, `init` suggests to be called only once (which is fine in most cases), but I would expect then `resetExtremes` for the following calls instead of `initExtremes()`.

Maybe we should simply pick a separate name `getColor`/`getSymbol` and `getExtremes`? Like this:
- `initColor`/`initSymbol`
- `computeExtremes` 

I think `resolveSomething` is a bad idea, suggests kind of relation with `Promise.resolve()`, which is not related to the Promises itself.

----
From a different perspective, maybe we should stick to `getSomething`, like this:
- `this.symbol = this.getSymbol()`
- `this.color = this.getColor()`
- `this.extremes = this.getExtremes()`

We already have this solution for `series.name = series.getName()`.

**Comment by bre1470** [comment]
I like the different perspective of @pawelfus. It avoids renaming and fulfills expectations. We could also combine it in the way, that it returns the value, but right before also set the property on the instance like it does now.

**Comment by bre1470** [comment]
@TorsteinHonsi Can we agree on changes for 7.1 or is it something for later?

**Comment by TorsteinHonsi** [comment]
Going back to start:

> I want the Serie color by calling `getColor()`, the Serie symbol by calling `getSymbol()`

Those functions are private and shouldn't be exposed in TypeScript. The mentioned properties are available as `series.color` and `series.symbol`. 

Apart from that, yes we should rename the functions but it doesn't affect the API.

**Comment by bre1470** [comment]
In that case we could postpone it for the time after ES6. I will fix the problem for now by marking the following members private: `getChartSize`, `getColor`, `getExtremes`, `getSymbol`.

**Comment by TorsteinHonsi** [comment]
Yes those are all private.
